aus400/regrid.py

```diff
@@ -144,3 +152,67 @@ def to_barra(data: xarray.Dataset):
     weights = xarray.open_dataset(root / "grids" / f"weights_{grid}_to_barrat.nc")
 
     return regrid(data, weights=weights)
+
+
+
+def regrid_vector(data):
+    """
+    Redrigs vector quantities like u/v defined on grid edges to the
+    scalar grid on gridpoint centres (same resolution as original)
+
+    will need to load a 'dummy' variable to get the coordinates
+    of the gridpoint centres at the same resolution as input data
+    - might be able to just use grids in /g/data instead?
+    -- > these are actually different from the variable grids (?)
+
+    Inputs:
+        data: input data
+    Outputs:
+        data_regrid: regridded data
+    """
+
+    res = identify_resolution(data)        
+
+    # load the grid to reshape input data to
+    # note: for now we are using pressure as a dummy variable to get the target grid
+    # for some reason, the d0198t/d0036t grids are very slightly different from the
+    # actual grid for these variables (offest by around 10e-6)
+
+
+    # grid_path = '/g/data/ia89/aus400/grids/'
+    # if res == 'd0036':
+    #     grid = xr.open_dataarray(grid_path+'d0036t.nc')
+    # elif res == 'd0198':
+    #     grid = xr.open_dataarray(grid_path+'d0198t.nc')
+
+    # note: dims are called lat/lon on the grid but latitude/longitude on the input
+    # change this to be consistent
+    #grid = grid.rename({'lat': 'latitude', 'lon': 'longitude'})
+
+    grid = load(
+        resolution=res,
+        stream="mdl",
+        variable="pressure",
+        time=slice(
+            pandas.offsets.Hour().rollback(data["time"].values[0])
+            - pandas.offsets.Hour(),
+            pandas.offsets.Hour().rollback(data["time"].values[-1])
+            + pandas.offsets.Hour(),
+        ),
+        ensemble=slice(data["ensemble"].values[0], data["ensemble"].values[-1]),
+    )["pressure"]    
```

Comment:
If the 'grids' files aren't working it might be better to use one of the 'fx' variables, rather than loading a model level variable for multiple times